src/wmtk/attribute/AttributeManager.hpp

```diff
@@ -137,6 +142,21 @@ MeshAttributeHandle<T> AttributeManager::register_attribute(
     return r;
 }
 template <typename T>
+inline T AttributeManager::parent_scope(std::function<T()> f)
+{
+    change_to_parent_scope();
+    T return_value = f();
```

Comment:
I'm not sure we need to make sure everything passes into a std::function. in reality we just want something that satisfies the `std::is_invokable` concept (https://en.cppreference.com/w/cpp/concepts/invocable). Too bad we're not on c++20, but i think it might be cleaner to static assert for invokability rather than to force everything get mapped to a std::function